website/frontend/src/components/blocks/Wallet/Wallet.tsx

```diff
@@ -1,31 +1,134 @@
 import React, { useEffect, useState } from 'react';
-import { useAlert } from 'react-alert';
-import { toShortAddress, copyToClipboard } from '../../../helpers';
+import { web3Accounts, web3Enable } from '@polkadot/extension-dapp';
+import { useDispatch, useSelector } from 'react-redux';
+import Identicon from '@polkadot/react-identicon';
+import { LogoutIcon } from 'assets/Icons';
+import { RootState } from 'store/reducers';
+import { setCurrentAccount, resetCurrentAccount } from 'store/actions/actions';
+import { UserAccount } from '../../../types/account';
+import { useApi } from '../../../hooks/useApi';
+import { Modal } from '../Modal';
+import { AccountList } from '../AccountList';
 
 import './Wallet.scss';
 
 export const Wallet = () => {
-  const [address, setAddress] = useState('');
+  const [ingectedAccounts, setIngectedAccounts] = useState<Array<UserAccount> | null>(null);
+  const [accountBalance, setAccountBalance] = useState<string | null>(null);
+  const [isOpen, setIsOpen] = useState(false);
 
-  const alert = useAlert();
+  const [api] = useApi();
+  const dispatch = useDispatch();
+  const currentAccount = useSelector((state: RootState) => state.account.account);
 
   useEffect(() => {
-    const publicKey: any = localStorage.getItem('public_key');
-    setAddress(publicKey);
-  }, [address]);
+    const getAllAccounts = async () => {
+      const extensions = await web3Enable('Gear Tech');
 
-  /* eslint-disable jsx-a11y/click-events-have-key-events, jsx-a11y/no-noninteractive-element-interactions */
+      // if extansion does not exist
+      if (extensions.length === 0) {
+        return;
+      }
+      const allAccounts = await web3Accounts();
+
+      allAccounts.forEach((acc: UserAccount) => {
+        if (acc.address === localStorage.getItem('savedAccount')) {
+          acc.isActive = true;
+          dispatch(setCurrentAccount(acc));
+        }
+      });
+      setIngectedAccounts(allAccounts);
+    };
+
+    // TODO: FIND ANOTHER WAY BE SHORE THAT EXTENSION IS READY
+    setTimeout(() => {
+      getAllAccounts();
+    }, 100);
+  }, [dispatch]);
+
+  // Get free balance for the chosen account
+
+  useEffect(() => {
+    const getBalance = async (ADDR: string) => {
+      const freeBalance = await api!.balance.findOut(ADDR);
+      setAccountBalance(freeBalance.toHuman());
+    };
+
+    if (currentAccount && api) {
+      getBalance(currentAccount.address);
+    }
+  }, [currentAccount, api, dispatch]);
+
+  const toggleModal = () => {
+    setIsOpen(!isOpen);
+  };
+
+  useEffect(() => {
+    if (isOpen) document.body.style.overflow = 'hidden';
```

Comment:
we could try to use this one https://github.com/willmcpo/body-scroll-lock#reactes6